edb/lib/std/30-jsonfuncs.edgeql

```diff
@@ -64,201 +68,279 @@ std::json_get(
 
 
 CREATE INFIX OPERATOR
-std::`=` (l: std::json, r: std::json) -> std::bool
+std::`=` (l: std::json, r: std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR r'=';
+};
 
 
 CREATE INFIX OPERATOR
-std::`?=` (l: OPTIONAL std::json, r: OPTIONAL std::json) -> std::bool
+std::`?=` (l: OPTIONAL std::json, r: OPTIONAL std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL EXPRESSION;
+};
 
 
 CREATE INFIX OPERATOR
-std::`!=` (l: std::json, r: std::json) -> std::bool
+std::`!=` (l: std::json, r: std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR r'<>';
+};
 
 
 CREATE INFIX OPERATOR
-std::`?!=` (l: OPTIONAL std::json, r: OPTIONAL std::json) -> std::bool
+std::`?!=` (l: OPTIONAL std::json, r: OPTIONAL std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL EXPRESSION;
+};
 
 
 CREATE INFIX OPERATOR
-std::`>=` (l: std::json, r: std::json) -> std::bool
+std::`>=` (l: std::json, r: std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR '>=';
+};
 
 
 CREATE INFIX OPERATOR
-std::`>` (l: std::json, r: std::json) -> std::bool
+std::`>` (l: std::json, r: std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR '>';
+};
 
 
 CREATE INFIX OPERATOR
-std::`<=` (l: std::json, r: std::json) -> std::bool
+std::`<=` (l: std::json, r: std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR '<=';
+};
 
 
 CREATE INFIX OPERATOR
-std::`<` (l: std::json, r: std::json) -> std::bool
+std::`<` (l: std::json, r: std::json) -> std::bool {
+    SET volatility := 'IMMUTABLE';
     FROM SQL OPERATOR '<';
+};
 
 
 ## CASTS
 
 # This is only a container cast, and subject to element type cast
 # availability.
-CREATE CAST FROM array<anytype> TO std::json
+CREATE CAST FROM array<anytype> TO std::json {
+    SET volatility := 'STABLE';
     FROM SQL FUNCTION 'to_jsonb';
+};
 
 
 # This is only a container cast, and subject to element type cast
 # availability.
-CREATE CAST FROM anytuple TO std::json
+CREATE CAST FROM anytuple TO std::json {
+    SET volatility := 'STABLE';
     FROM SQL EXPRESSION;
+};
 
 
-CREATE CAST FROM std::json TO array<json>
+CREATE CAST FROM std::json TO array<json> {
+    SET volatility := 'IMMUTABLE';
     FROM SQL $$
         SELECT array_agg(j)
         FROM jsonb_array_elements(val) AS j
     $$;
+};
 
 
-CREATE CAST FROM std::bool TO std::json
+CREATE CAST FROM std::bool TO std::json {
+    SET volatility := 'STABLE';
```

Comment:
Just curious, why isn't this immutable?

Reply:
Because `to_jsonb` isn't immutable.

Reply:
Although it's worth investigating exactly why and for which specific types, since `to_jsonb()` is generic.

Reply:
I suggest to add comments then to every function that isn't IMMUTABLE explaining why.

Reply:
+1

Reply:
I've updated all the schema definitions to have volatility specified explicitly - this would make it easier to read and not have to guess what the default is.

I've added more comments about the chosen volatility value.